project_stock/tests/test_project_stock.py

```diff
@@ -51,121 +58,130 @@ def test_project_task_misc(self):
         self.assertEqual(self.move_product_b.raw_material_task_id, self.task)
 
     def _test_task_analytic_lines_from_task(self, amount):
-        self.assertEqual(len(self.task.stock_analytic_line_ids), 2)
-        self.assertEqual(
-            sum(self.task.stock_analytic_line_ids.mapped("unit_amount")), 3
-        )
-        self.assertEqual(
-            sum(self.task.stock_analytic_line_ids.mapped("amount")), amount
-        )
+        task = self.env["project.task"].browse(self.task.id)
+        self.assertEqual(len(task.stock_analytic_line_ids), 2)
+        self.assertEqual(sum(task.stock_analytic_line_ids.mapped("unit_amount")), 3)
+        self.assertEqual(sum(task.stock_analytic_line_ids.mapped("amount")), amount)
         self.assertEqual(
-            self.task.stock_analytic_tag_ids,
-            self.task.stock_analytic_line_ids.mapped("tag_ids"),
+            task.stock_analytic_tag_ids,
+            task.stock_analytic_line_ids.mapped("tag_ids"),
         )
         self.assertIn(
             self.analytic_account,
-            self.task.stock_analytic_line_ids.mapped("account_id"),
+            task.stock_analytic_line_ids.mapped("account_id"),
         )
 
+    @users("manager-user")
```

Comment:
Do both with admin and this user

Reply:
What about this?: Create another method `test_..._admin` with the other @users decorator and calling the first method